service/test/controllers/organisationmanagement/OrganisationControllerTest.java

```diff
@@ -1,404 +1,163 @@
-    innerMap.put(JsonKey.USER_ID, "userId");
-    requestMap.put(JsonKey.REQUEST, innerMap);
-    String data = mapToJson(requestMap);
-
-    JsonNode json = Json.parse(data);
-    RequestBuilder req =
-        new RequestBuilder().bodyJson(json).uri("/v1/org/member/add").method("POST");
-    req.headers(headerMap);
-    Result result = route(req);
-    assertEquals(200, result.status());
+  public void testGetOrgDetailsSuccess() {
+    Result result =
+        performTest(
+            "/v1/org/read",
+            "POST",
+            createOrganisationRequest(null, orgId, false, null, status, null));
+    assertEquals(getResponseCode(result), ResponseCode.success.getErrorCode().toLowerCase());
+    assertTrue(getResponseStatus(result) == 200);
   }
 
   @Test
-  public void testremoveMemberFromOrganisation() {
-    PowerMockito.mockStatic(RequestInterceptor.class);
-    when(RequestInterceptor.verifyRequestData(Mockito.anyObject()))
-        .thenReturn("{userId} uuiuhcf784508 8y8c79-fhh");
-    Map<String, Object> requestMap = new HashMap<>();
-    Map<String, Object> innerMap = new HashMap<>();
-    innerMap.put(JsonKey.ORGANISATION_ID, "org123");
-    innerMap.put(JsonKey.USER_ID, "userId");
-    requestMap.put(JsonKey.REQUEST, innerMap);
-    String data = mapToJson(requestMap);
-
-    JsonNode json = Json.parse(data);
-    RequestBuilder req =
-        new RequestBuilder().bodyJson(json).uri("/v1/org/member/remove").method("POST");
-    req.headers(headerMap);
-    Result result = route(req);
-    assertEquals(200, result.status());
+  public void testGetOrgDetailsFailureWithoutOrgId() {
+    Result result =
+        performTest(
+            "/v1/org/read",
+            "POST",
+            createOrganisationRequest(null, null, false, null, status, null));
+    assertEquals(getResponseCode(result), ResponseCode.mandatoryParamsMissing.getErrorCode());
+    assertTrue(getResponseStatus(result) == 400);
   }
 
   @Test
-  public void testjoinUserOrganisation() {
-    PowerMockito.mockStatic(RequestInterceptor.class);
-    when(RequestInterceptor.verifyRequestData(Mockito.anyObject()))
-        .thenReturn("{userId} uuiuhcf784508 8y8c79-fhh");
-    Map<String, Object> requestMap = new HashMap<>();
-    Map<String, Object> innerMap = new HashMap<>();
-    innerMap.put(JsonKey.ORGANISATION_ID, "org123");
-    innerMap.put(JsonKey.USER_ID, "userId");
-    requestMap.put(JsonKey.REQUEST, innerMap);
-    String data = mapToJson(requestMap);
-
-    JsonNode json = Json.parse(data);
-    RequestBuilder req =
-        new RequestBuilder().bodyJson(json).uri("/v1/org/member/join").method("POST");
-    req.headers(headerMap);
-    Result result = route(req);
-    assertEquals(200, result.status());
+  public void testSearchOrgSuccess() {
+    Result result =
+        performTest(
+            "/v1/org/search",
+            "POST",
+            createOrganisationRequest(null, null, false, null, status, new HashMap<>()));
```

Comment:
Why search org is using createOrganisationRequest? This should be separate if at all.